### ISSUE TITLE
Random Process Seeding with Torch

### ISSUE THREAD

**Issue by roddtalebi** [ezCGP/ezCGP#227]
It doesn't seem like seeding is working with simgan/pytorch...don't know where it broke but I'm getting different losses whenever I rerun ezcgp with '-s 9' in the command

**Comment by roddtalebi** [comment]
Added this code in main.py to test/verify that random-process-seeding not working for pytorch 
```
    # set the seed before importing problem.
    # NOTE will set another seed when we start the universe
    ezLogging.warning("Setting seed, for file imports, to %i" % (seed))
    np.random.seed(seed)
    random.seed(seed) # shouldn't be using 'random' module but setting seed jic

    #####
    import torch
    faker_py = torch.randn((2,5))
    faker_np = np.random.random((2,5))
    print(faker_py)
    print(faker_np)
    exit()
```

Run twice with ```python main.py -p poop -s 9``` and got ...
```
Setting seed, for file imports, to 9
tensor([[-0.6322,  0.7001, -0.0240,  0.1793, -0.9340],
        [ 1.7536,  0.7126, -0.6603,  0.6382, -0.1913]])
[[0.01037415 0.50187459 0.49577329 0.13382953 0.14211109]
 [0.21855868 0.41850818 0.24810117 0.08405965 0.34549864]]
0.46300735781502145
```
and then
```
Setting seed, for file imports, to 9
tensor([[ 0.9652,  1.2294,  0.2441, -0.0653, -0.8203],
        [-0.0793, -1.5823, -0.1406, -0.0375, -1.7593]])
[[0.01037415 0.50187459 0.49577329 0.13382953 0.14211109]
 [0.21855868 0.41850818 0.24810117 0.08405965 0.34549864]]
0.46300735781502145
```

So numpy and random modules are seeded correctly...I guess pytorch is either imported somewhere earlier in the code or have to set it manually

**Comment by roddtalebi** [comment]
Fixed it by adding this into main right after it sets the seed for numpy and random modules
```
    try:
        import tensorflow as tf
        tf.random.set_seed(seed)
    except:
        pass
    try:
        import torch
        torch.manual_seed(seed)
        torch.use_deterministic_algorithms(True)
    except:
        pass
```

**Comment by roddtalebi** [comment]
might be a gpu random process thing...

if it makes us feel better, try running on a cpu and see if that 'fixes the issue'
